### PR TITLE
Change root console handling for remote backends

### PR DESCRIPTION
**Pull request by DrMullings**
As discussed in poo#45146

- Related ticket: https://progress.opensuse.org/issues/45146

### CODE REVIEW

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -16,7 +16,7 @@ use utils qw(
 );
 use version_utils qw(is_hyperv_in_gui is_sle is_leap is_svirt_except_s390x is_tumbleweed is_opensuse);
 use x11utils qw(desktop_runner_hotkey ensure_unlocked_desktop);
-use Utils::Backends 'use_ssh_serial_console';
+use Utils::Backends 'use_ssh_serial_console', 'is_remote_backend';
```
Comment (foursixnine) [inline lib/susedistribution.pm]:
```suggestion
use Utils::Backends qw(use_ssh_serial_console is_remote_backend);
```

File: tests/sles4sap/hana_install.pm [hunk 2]
```diff
@@ -32,7 +32,7 @@ sub run {
     my $password = 'Qwerty_123';
     set_var('PASSWORD', $password);
 
-    check_var('BACKEND', 'ipmi') ? use_ssh_serial_console : select_console 'root-console';
+    select_console 'root-console';
```
Comment (foursixnine) [inline tests/sles4sap/hana_install.pm]: @alvarocarvajald heads up

File: lib/susedistribution.pm (hunk 1 above)
Comment (DrMullings) [reply lib/susedistribution.pm]: Done

**Comment by alvarocarvajald** [comment]
LGTM. I'd like to see some verification runs though.

**Comment by DrMullings** [comment]
@alvarocarvajald 
> LGTM. I'd like to see some verification runs though.

As I have no clue of the SAP modules, can I just take the newest SLE12 build and SCHEDULE= the changed modules?

**Comment by alvarocarvajald** [comment]
> As I have no clue of the SAP modules, can I just take the newest SLE12 build and SCHEDULE= the changed modules?

You will also need to provide installation media for the SAP product(s), as well as values for `INSTANCE_ID`, `INSTANCE_SID` and `INSTANCE_TYPE` IIRC. I suppose you could clone https://openqa.suse.de/tests/3111556 and fire it on a different IPMI worker. As long as `sles4sap/patterns` at least attempts to install the SAP patterns, i.e., it has access to a root console, I think it should be enough.

**Comment by DrMullings** [comment]
Started https://openqa.suse.de/tests/3115727# as verification

**Comment by DrMullings** [comment]
Sadly I cannot verify it as long as there is no working scenario that reaches these modules

**Comment by SergioAtSUSE** [comment]
@alvarocarvajald, is there any "passing" job we can use to verify?

If not, I would consider this PR can be merged.

**Comment by alvarocarvajald** [comment]
> @alvarocarvajald, is there any "passing" job we can use to verify?
> 
> If not, I would consider this PR can be merged.

Not AFAICS. All of the IPMI jobs I looked for are failing in `installation/first_boot`. I guess we can merge.

**Comment by SergioAtSUSE** [comment]
@DrMullings, please, give some love to travis: https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/562010044#L662

**Comment by DrMullings** [comment]
@SergioAtSUSE if you take a look at the output Travis complains about Line#380 which I did not touch, shall I manually put it as it wants it or can we ignore it then?

**Comment by pevik** [comment]
@DrMullings Hm, commit message `As discussed in poo#45146`, IMHO it's better to add short overview (why to follow whole discussion when you browse git history; also people sometimes work offline).

**Comment by pevik** [comment]
@DrMullings does it mean that `use_ssh_serial_console` is not needed in tests? Can I remove it from kernel tests?
tests/kernel/boot_ltp.pm
tests/kernel/install_ltp.pm
tests/kernel/update_kernel.pm

**Comment by DrMullings** [comment]
@pevik yes that was the intention of that PR ;)

**Pull request merged**